rbs/SigsRewriter.cc

```diff
@@ -382,6 +205,111 @@ void insertHelpers(unique_ptr<parser::Node> *body, parser::NodeVec helpers) {
 
 } // namespace
 
+Comments SigsRewriter::commentsForNode(core::MutableContext ctx, parser::Node *node) {
+    auto comments = Comments{};
+    enum class SignatureState { None, Started, Multiline };
+    auto state = SignatureState::None;
+
+    if (auto commentsNodesEntry = commentsByNode.find(node); commentsNodesEntry != commentsByNode.end()) {
+        auto commentsNodes = commentsNodesEntry->second;
+        auto declaration_comments = std::vector<Comment>();
+
+        for (auto &commentNode : commentsNodes) {
+            // If the comment starts with `# @`, it's an annotation
+            if (absl::StartsWith(commentNode.string, "# @")) {
+                auto comment = Comment{
+                    .commentLoc = commentNode.loc,
+                    .typeLoc = core::LocOffsets{commentNode.loc.beginPos() + 3, commentNode.loc.endPos()},
+                    .string = commentNode.string.substr(3),
+                };
+
+                comments.annotations.emplace_back(move(comment));
+                continue;
+            }
+
+            // If the comment starts with `#:`, it's a signature
+            if (absl::StartsWith(commentNode.string, "#:")) {
+                if (state == SignatureState::Multiline) {
+                    if (auto e = ctx.beginError(commentNode.loc, core::errors::Rewriter::RBSMultilineMisformatted)) {
+                        e.setHeader("Signature start (\"#:\") cannot appear after a multiline signature (\"#|\")");
+                        return comments;
```

Comment:
I chose to early return for both of these `RBSMultilineMisformatted` error cases. When we keep going and translate there are more error messages that may be ambiguous.